controllers/dnspolicy_controller_multi_cluster_test.go

```diff
@@ -45,8 +48,21 @@ var _ = Describe("DNSPolicy Multi Cluster", func() {
 		gatewayClass = testBuildGatewayClass("gwc-"+testNamespace, "default", "kuadrant.io/bar")
 		Expect(k8sClient.Create(ctx, gatewayClass)).To(Succeed())
 
-		managedZone = testBuildManagedZone("mz-example-com", testNamespace, "example.com")
+		dnsProviderSecret = testBuildInMemoryCredentialsSecret("inmemory-credentials", testNamespace)
+		managedZone = testBuildManagedZone("mz-example-com", testNamespace, "example.com", dnsProviderSecret.Name)
+		Expect(k8sClient.Create(ctx, dnsProviderSecret)).To(Succeed())
 		Expect(k8sClient.Create(ctx, managedZone)).To(Succeed())
+		Eventually(func(g Gomega) {
+			err := k8sClient.Get(ctx, client.ObjectKeyFromObject(managedZone), managedZone)
+			g.Expect(err).NotTo(HaveOccurred())
+			g.Expect(managedZone.Status.Conditions).To(
+				ContainElement(MatchFields(IgnoreExtras, Fields{
+					"Type":               Equal(string(kuadrantdnsv1alpha1.ConditionTypeReady)),
+					"Status":             Equal(metav1.ConditionTrue),
+					"ObservedGeneration": Equal(managedZone.Generation),
+				})),
+			)
+		}, TestTimeoutMedium, time.Second).Should(Succeed())
```

Comment:
Note: This will now expect the dns controller to be running on the target cluster. Up to now the dns policy integration tests here have not required that. If this becomes an issue we can remove the status check.

An alternative might be to only add status checks like this in the specs we want to do it in and label them as requiring a running dns operator on cluster allowing them to be excluded if needs be. However, RLP and Auth policy tests both also require operators to be running in order to create the kuadrant CR so we already have a requirement for running sub operators in the integration test suite anyway.